solc/CommandLineInterface.cpp

```diff
@@ -610,6 +614,23 @@ Allowed options)",
 
 bool CommandLineInterface::processInput()
 {
+	if (m_args.count(g_argStandardJSON))
+	{
+		string input;
+		while (!cin.eof())
```

Comment:
You are right, my mistake, there are only functions for files. `cin` is not opened in binary mode, so I think it will convert `\r\n` into `\n`. But actually, this should be irrelevant, since everything is json-encoded and whitespace should be just ignored anyway. So, yes, just leave it like that!